x-pack/solutions/observability/plugins/streams_app/public/components/stream_detail_lifecycle/hooks/use_wired_streams.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useEffect, useState } from 'react';
+import {
+  StreamGetResponse,
+  WiredStreamDefinition,
+  isWiredStreamDefinition,
+  isWiredStreamGetResponse,
+} from '@kbn/streams-schema';
+import { useAbortController } from '@kbn/observability-utils-browser/hooks/use_abort_controller';
+import { useKibana } from '../../../hooks/use_kibana';
+
+export const useWiredStreams = ({ definition }: { definition: StreamGetResponse }) => {
+  const {
+    dependencies: {
+      start: {
+        streams: { streamsRepositoryClient },
+      },
+    },
+  } = useKibana();
+  const { signal } = useAbortController();
```

Comment:
why not using `useStreamsAppFetch` here? Seems like it could slightly simplify

Reply:
nice one, fixed in https://github.com/elastic/kibana/pull/210055/commits/12ef014229863d3180243c2fb05c5628222fed54